src/Raven.Server/Documents/Revisions/RevisionsBinCleanMergedCommand.cs

```diff
@@ -0,0 +1,101 @@
+﻿using System;
+using System.Collections.Generic;
+using Raven.Server.Documents.TransactionMerger.Commands;
+using Raven.Server.ServerWide.Context;
+using Voron;
+
+namespace Raven.Server.Documents.Revisions;
+
+public partial class RevisionsStorage
+{
+    internal sealed class RevisionsBinCleanMergedCommand : DocumentMergedTransactionCommand
+    {
+        private List<(string Id, long Etag)> _idsAndEtags;
+        private readonly long _lastEtag;
+        private readonly bool _isFirst;
+
+        private const int MaxDeletesUponUpdate = 1024;
+
+        public (int DeletedEntries, int StartIndex) Result { get; private set; }
+
+        public RevisionsBinCleanMergedCommand(List<(string Id, long Etag)> idsAndEtags, long lastEtag, bool isFirst)
+        {
+            _idsAndEtags = idsAndEtags;
+            _lastEtag = lastEtag;
+            _isFirst = isFirst;
+        }
+
+        protected override long ExecuteCmd(DocumentsOperationContext context)
+        {
+            Result = DeleteRevisions(context);
+            var etag = Result.StartIndex == _idsAndEtags.Count ? _lastEtag : _idsAndEtags[Result.StartIndex].Etag;
+            SetLastRevisionsBinCleanerLastEtag(context, etag);
+            return 1;
+        }
+
+        private (int DeletedEntries, int StartIndex) DeleteRevisions(DocumentsOperationContext context)
+        {
+            var revisionsStorage = context.DocumentDatabase.DocumentsStorage.RevisionsStorage;
+
+            var index = 0;
+            var deletedEntities = 0;
+
+            foreach (var (id, etag) in _idsAndEtags)
+            {
+                using (DocumentIdWorker.GetSliceFromId(context, id, out Slice lowerId))
+                using (revisionsStorage.GetKeyPrefix(context, lowerId, out Slice prefixSlice))
+                {
+                    var collectionName = revisionsStorage.GetCollectionFor(context, prefixSlice);
+                    if (collectionName == null)
+                    {
+                        if (revisionsStorage._logger.IsInfoEnabled)
+                            revisionsStorage._logger.Info($"Tried to delete revisions for '{id}' but no revisions found.");
+                        continue;
```

Comment:
should you do `index++` here?

Reply:
Done (added 'else' instead of `index++` twice